tests/Silverback.Integration.Tests.E2E/Mqtt/EncryptionTests.cs

```diff
@@ -78,6 +78,173 @@ public async Task Encryption_SimpleMessages_EncryptedAndDecrypted()
+        {
+            var message1 = new TestEventOne { Content = "Message 1" };
+            var message2 = new TestEventOne { Content = "Message 2" };
+            const string expectedKeyIdentifier = "my-encryption-key-id";
+
+            Host.ConfigureServices(
+                    services => services
+                        .AddLogging()
+                        .AddSilverback()
+                        .UseModel()
+                        .WithConnectionToMessageBroker(options => options.AddMockedMqtt())
+                        .AddMqttEndpoints(
+                            endpoints => endpoints
+                                .Configure(
+                                    config => config
+                                        .WithClientId("e2e-test")
+                                        .ConnectViaTcp("e2e-mqtt-broker"))
+                                .AddOutbound<IIntegrationEvent>(
+                                    endpoint => endpoint
+                                        .ProduceTo(DefaultTopicName)
+                                        .EncryptUsingAes(AesEncryptionKey, null, expectedKeyIdentifier))
+                                .AddInbound(
+                                    endpoint => endpoint
+                                        .ConsumeFrom(DefaultTopicName)
+                                        .DecryptUsingAes(AesEncryptionKey, _ => AesEncryptionKey)))
+                        .AddIntegrationSpyAndSubscriber())
+                .Run();
+
+            var publisher = Host.ScopedServiceProvider.GetRequiredService<IEventPublisher>();
+            await publisher.PublishAsync(message1);
+            await publisher.PublishAsync(message2);
+
+            await Helper.WaitUntilAllMessagesAreConsumedAsync();
+
+            Helper.Spy.OutboundEnvelopes.Should().HaveCount(2);
+            Helper.Spy.OutboundEnvelopes[0].RawMessage.Should().BeOfType<SymmetricEncryptStream>();
+            Helper.Spy.OutboundEnvelopes[1].RawMessage.Should().BeOfType<SymmetricEncryptStream>();
+            Helper.Spy.OutboundEnvelopes.ForEach(
+                envelope =>
+                    envelope.Headers.GetValue(DefaultMessageHeaders.EncryptionKeyId).Should()
+                        .Be(expectedKeyIdentifier));
+
+            Helper.Spy.InboundEnvelopes.Should().HaveCount(2);
+            Helper.Spy.InboundEnvelopes[0].Message.Should().BeEquivalentTo(message1);
+            Helper.Spy.InboundEnvelopes[1].Message.Should().BeEquivalentTo(message2);
+            Helper.Spy.InboundEnvelopes.ForEach(
+                envelope =>
+                    envelope.Headers.GetValue(DefaultMessageHeaders.EncryptionKeyId).Should()
+                        .Be(expectedKeyIdentifier));
+
+            Helper.GetClientSession("e2e-test").PendingMessagesCount.Should().Be(0);
+        }
+
+        [Fact]
+        public async Task Encryption_NoCallback_DefaultEncryptionKeyIsUsed()
```

Comment:
Isn't this the normal case? Wasn't that tested already?

If the intention is to test that it still works, despite the header being passed, call it accordingly (something like `Encryption_KeyHeaderWithFixedDecryptionKey` or similar), otherwise I'll never remember for what it is... 😝

Reply:
😀 this test is the normal case, I deleted it and just added in the original one the check of the custom header (that has to be null).